ads/google/a4a/utils.js

```diff
@@ -357,92 +356,71 @@ export function additionalDimensions(win, viewportSize) {
 /**
  * Extracts configuration used to build amp-analytics element for active view.
  *
+ * @param {!../../../extensions/amp-a4a/0.1/amp-a4a.AmpA4A} a4a
  * @param {!../../../src/service/xhr-impl.FetchResponseHeaders} responseHeaders
  *   XHR service FetchResponseHeaders object containing the response
  *   headers.
- * @param {!../../../src/service/extensions-impl.Extensions} extensions
- * @return {?AmpAnalyticsConfigDef} config or null if invalid/missing.
+ * @return {?JSONType} config or null if invalid/missing.
  */
-export function extractAmpAnalyticsConfig(responseHeaders, extensions) {
+export function extractAmpAnalyticsConfig(a4a, responseHeaders) {
   if (responseHeaders.has(AMP_ANALYTICS_HEADER)) {
     try {
       const analyticsConfig =
         JSON.parse(responseHeaders.get(AMP_ANALYTICS_HEADER));
-      dev().assert(Array.isArray(analyticsConfig['url']));
-      if (analyticsConfig.url.length) {
-        extensions.loadExtension('amp-analytics');
+      console.log('anayticsConfig is ', analyticsConfig['url']);
+      dev().assert(Array.isArray(analyticsConfig['url']) &&
+          analyticsConfig['url'].length);
+      const urls = analyticsConfig.url;
+
+      const visibilitySpec = {
+        'selector': 'amp-ad',
+        'selectionMethod': 'closest',
+        'visiblePercentageMin': 50,
+        'continuousTimeMin': 1000,
+      };
+      const config = {
+        'transport': {'beacon': false, 'xhrpost': false},
+        'triggers': {
+          'continuousVisible': {
+            'on': 'visible',
+            visibilitySpec,
+          },
+          'continuousVisibleIniLoad': {
+            'on': 'ini-load',
+            visibilitySpec,
+          },
+          'continuousVisibleRenderStart': {
+            'on': 'render-start',
+            visibilitySpec,
```

Comment:
I realize you're just carrying over the existing functionality but does the visibilitySpec given actually make sense for render-start or ini-load?  Seems like all they really need are their request values.  Mind fixing?  You could then just next visibilitySpec within the config only for continuousVisible trigger